articles/javascript-higher-order-functions/index.md

```diff
@@ -0,0 +1,333 @@
+
+Functional programming is a significant building block for JavaScript as a language. They are some of the common scrips that you will continuously use when building JavaScript applications. Functions set statements to perform tasks. Functions define a task, which you can call and execute whenever you want inside your program.
+
+JavaScript treats a function as an object assigned to values to return a value. They have methods and properties, just like the other objects. Unlike ordinary objects, a function can be
+
+- Called.
+- Passed as an argument to another function.
+- Assigned to a variable as a value.
+- Be returned by another function.
+
+Functions are fundamental that outside the "normal" functions, JavaScript has high order functions.
+
+A high order function accepts another function as an argument or returns a function as its output. The concept is commonly referred to as a callback function. A function that takes another function as its argument is what we refer to as a callback function because it calls back that function.
```

Comment:
Callback functions are important concepts and used extensively in React(I have worked with Callbacks in React). Can you explain this in detail, giving a small example? The example may or may not be language-specific.